lib/java/src/org/apache/thrift/transport/TInvalidFrameSizeException.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.thrift.transport;
+
+public class TInvalidFrameSizeException extends TTransportException {
```

Comment:
I always understood Thrift as being as consistent as possible across languages, and I think this is a benefit that we should not give up. I already said that earlier, but I say it again: Introducing a new exception typ is nothing I'm against per se, quite the contrary. But if we do it only in one language, we start to break this convention and open doors that we don't want open. [Recommended reading: Broken Window theory](https://en.wikipedia.org/wiki/Broken_windows_theory).